src/Plot.js

```diff
@@ -477,17 +481,16 @@ export default class Plot extends Viz {
     const xOffset = width - this._xTest._getRange()[1];
     const xDifference = xOffsetRight - xOffset + this._xTest.padding();
 
-    const x2Offset = width - this._x2Test._getRange()[1];
+    const x2Offset = this._x2Test._getRange()[1] !== undefined ? width - this._x2Test._getRange()[1] : width;
     const x2Difference = xOffsetRight - x2Offset + this._x2Test.padding();
-
     const xBounds = this._xTest.outerBounds();
     const xHeight = xBounds.height + this._xTest.padding();
 
     const yOffsetBottom = max([xHeight, height - this._yTest._getRange()[1], height - this._y2Test._getRange()[1]]);
     const yAxisOffset = height - this._yTest._getRange()[1];
     const yDifference = isYAxisOrdinal ? yOffsetBottom - yAxisOffset + this._yTest.padding() : xHeight;
 
-    const y2AxisOffset = height - this._y2Test._getRange()[1];
+    const y2AxisOffset = this._y2Test._getRange()[1] !== undefined ? height - this._y2Test._getRange()[1] : height;
```

Comment:
We might see a very small speed increase by not calling the `_getRange()` functions 3 times for each axis. Store them (all 4 axes) in local variables first like this:

```js
const y2Range = this._y2Test._getRange();
```